Section 2/2.15_17_Working_with_Strings_Part1_&_Part2.rb

```diff
@@ -0,0 +1,76 @@
+# WORKING WITH STRINGS Part 1
+# String concatenation
+first_name = "hanyi"
+last_name = "koh"
+puts first_name + " " + last_name
+full_name = first_name + last_name
+
+# String interpolation
+puts "My first name is #{first_name} and my last name is #{last_name}"
+
+# Using method
+puts full_name.class    
+# => String
+p 10.methods # find all the invokable methods for the object by default
+# => [:bit_length, :digits, :|, :gcd, :lcm, ...]
+p 10.class
+# => String
+## When using two methods on one object in the same line, is called method chain for example: 10.to_s.class
+p 10.to_s.class
+# => String
+
+p full_name.length       
+# => 8
+p full_name.reverse      
+# => "hokiynah"
+p full_name.capitalize # Only the first char will be capitalized, the other chars will be uncapitalized.   
+# => "Hanyikoh"
+p full_name.empty?       
+# => false
+p "".empty?              
+# => true
+p "".nil?                
+# => false (even though it is empty, it is not nil)
```

Comment:
Why?

Reply:
In Ruby, nil is a special value that denotes nothing or void or the absence of any value, it is an object of Nil Class. So the `.nil?` function will only return true if it is a Nil class object, "" is not.

`#empty? `can be used on some Ruby objects and collections including Arrays, Hashes and Sets. It returns true only if the object's length is zero or when they have no elements. Strings can be treated as a collection of bytes/characters, so when it contains no character, the length is zero therefore "" will return true

Reply:
Note key word: nil is a value of Nil class.